sdk/eventgrid/azure-eventgrid/azure/eventgrid/_signature_credential_policy.py

```diff
@@ -0,0 +1,23 @@
+import six
+
+from azure.core.pipeline.policies import SansIOHTTPPolicy
+
+class EventGridSharedAccessSignatureCredentialPolicy(SansIOHTTPPolicy):
```

Comment:
You don't need this class at all, use the AzureKeyCredentialPolicy instead:

https://github.com/Azure/azure-sdk-for-python/blob/2f7f59ef2dad9ea68104386eef81103599308b8b/sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py#L97

Reply:
I tried that originally, but in order to make that work, I would need the EventGridSharedAccessSignatureCredential to have a key property, rather than a signature property, since it sets the request header to `self._credential.key`, instead of `self._credential.signature`. I thought making that change would be confusing to the user, b/c they may pass in the key rather than generating the signature through `generate_shared_access_signature`.

Reply:
Note that I'm talking about the policy, not the credential class. If you wish the credential class to expose something called "signature", and then in map it to "key" in the internal implementation, you are still re-using the policy

Reply:
Design decision that needs to be revisited.